lib/plugins/package/lib/zip-service.js

```diff
@@ -218,20 +219,23 @@ async function excludeNodeDevDependencies(serviceDir) {
           ['dev', 'prod'],
           (env) => {
             const depFile = env === 'dev' ? nodeDevDepFile : nodeProdDepFile
-            return childProcess
-              .execAsync(
-                `npm ls --${env}=true --parseable=true --long=false --silent --all >> ${depFile}`,
-                {
-                  cwd: dirWithPackageJson,
-                  // We are overriding `NODE_ENV` because when it is set to "production"
-                  // it causes invalid output of `npm ls` with `--dev=true`
-                  env: {
-                    ...process.env,
-                    NODE_ENV: null,
-                  },
+            return execAsync(
+              `npm ls --${env}=true --parseable=true --long=false --silent --all >> ${depFile}`,
+              {
+                cwd: dirWithPackageJson,
+                // We are overriding `NODE_ENV` because when it is set to "production"
+                // it causes invalid output of `npm ls` with `--dev=true`
+                env: {
+                  ...process.env,
+                  NODE_ENV: null,
                 },
+              },
+            ).catch((err) => {
+              log.debug(
+                `Failed to run npm ls to retrieve dependencies for exclusion. Error: ${err.message}`,
               )
-              .catch(() => Promise.resolve())
+              Promise.resolve()
```

Comment:
This looks like a mistake introduced by the conversion of the `.catch` handler from an arrow function with an right-hand-side expression to an arrow function with braces. If you really want to return the `Promise.resolve()` you need to add an explicit `return`.

That said, `.catch` will (if its handler is executed) **already** resolve with whatever you return from its handler, which is `undefined` if you **don't** have explicit return, so you might as well drop this line entirely.

Reply:
You're right - thanks for pointing that out!